core/templates/rid_owner.h

```diff
@@ -351,6 +351,9 @@ class RID_Alloc : public RID_AllocBase {
 
 			for (size_t i = 0; i < max_alloc; i++) {
 				uint64_t validator = validator_chunks[i / elements_in_chunk][i % elements_in_chunk];
+				if (validator & 0x80000000) {
+					continue; //uninitialized
+				}
```

Comment:
About this change, @reduz said:
```
technically this situation should never happen so, while the fix is correct, I wonder what is causing it 
```

Reply:
`RID_Alloc::allocate_rid()` and `RID_Alloc::initialize_rid()` are two APIs, so this situation will happen if only the former is called.

In this case, the `initialize_rid()` inside `RendererStorageRD::texture_2d_initialize()` will be skipped if `ERR_FAIL_COND` is triggered. No error handling can be done in `texture_2d_create()` (the `FUNCRIDTEX1` macro).

https://github.com/godotengine/godot/blob/7a85e95a4b699960c540bb583cec526ab1b034d8/servers/rendering/renderer_rd/renderer_storage_rd.cpp#L543-L551

https://github.com/godotengine/godot/blob/7a85e95a4b699960c540bb583cec526ab1b034d8/servers/rendering/rendering_server_default.h#L156-L165

The parameter range validations in this PR are added to prevent this error condition from happening.